InkscapeExtension/ExtendedHatchFill.py

```diff
@@ -96,8 +101,109 @@ def recursivelyTraverseSvg(self, a_node_list):
+         cc = self.options.curv_coef # Curvature coefficient (steepness angle)                   
+         angle = math.radians(self.options.hatchAngle)
+
+          # absolute length of the segment between the points
+         totLen = ((x2 - x1) ** 2 + (y2 - y1) ** 2) ** 0.5          
+         w = x2 - x1 
+         h = y2 - y1 
+         wx = w / totLen # * math.cos(math.radians((angle)) # cos
+         hx = h / totLen # * math.sin(math.radians((angle)) # sin
+        
+         r1 = randint(-cc, cc)  
+         r2 = randint(-cc, cc)
+         r3 = randint(-cc, cc)
+         r4 = randint(-cc, cc)
+
+
+         if self.options.mode == "linear":
+            path = f'M {x1 + r1 + od * wx}, {y1 + r2 + od * hx} l {x2-x1 -r3 - 2 * od * wx}, {y2-y1 - r4 - 2 * od * hx}'
+         
+         if self.options.mode == "quadratic":
+            path = f'M {x1 + od * wx} {y1 + od * hx } \
+                     c {w / 3 + r1 * hx}, {h / 3 - r1 * wx} \
+                       {w / 3 - r2 * hx}, {h/ 3 + r2 * wx} \
+                       {x2 - x1 - 2 * od * wx} {y2 - y1 - 2* od * hx }'
+
+         if self.options.mode == "cubic":
+            path = f'M {x1 + od * wx} {y1 + od * hx } \
+                     q {w / 4 + 2 * r1 * hx}, {h / 4 + 2 * r1 * wx} \
+                       {w / 2 + r2 * hx}, {h / 2  + r2 * wx} \
+                     t {(x2 - x1)/ (2) - 2 * od * wx}, {(y2 - y1)/ (2)- 2* od * hx  }'
+
+
+         if self.options.mode == "hatch_line":
+             number_segm = self.options.number_segm
+             dx = (x2 - x1 - od * wx) / number_segm
+             dy = (y2 - y1 - od * hx) / number_segm
+             path = f'M {x1 + od * wx}, {y1 + od * hx} l {dx - r1 - od * wx} {dy - r2 - od * hx}'
+             for _ in range(number_segm - 1):
+                 r1 = randint(-cc, cc)
+                 r2 = randint(-cc, cc)
+                 x1 += dx
+                 y1 += dy
+                 path += f'M {x1}, {y1} l {dx - r1} {dy - r2}'
+
+
+         if self.options.mode == "hatch_curv":
+             path = ''
+             number_segm = self.options.number_segm
+             dx = (x2 - x1 - 2* od * wx) / number_segm
+             dy = (y2 - y1 - 2 * od * hx) / number_segm
+             for _ in range(number_segm):
+                 r1 = randint(-cc, cc) / 2
+                 r2 = randint(-cc, cc) / 2
+                 path += f'M {x1 + od * wx}, {y1 + od * hx} c {r1} , {r2} {-2 * r1},  {-2 * r2} {dx-1} {dy-1}'
+                 x1 += dx
+                 y1 += dy
+
+         if self.options.mode == "wavy":
+             path = ''
+             number_segm = self.options.number_segm
+             dx = (x2 - x1 - 2 * od * wx) / number_segm
+             dy = (y2 - y1 - 2 * od * hx) / number_segm
+             for _ in range(number_segm):
+                 r1 = randint(1, cc)
+                 r2 = randint(-cc, 0)
+                 path += f'M {x1 + od * wx}, {y1 + od * hx} \
+                             c {dx / 3}, {5} \
```

Comment:
what is 5? Is it a part of the formula or a constant you found empirically? Does it work for any values of wx, dx? Anyway, please write it without {}

Reply:
Hatch 'wavy line' in progress